src/bindings/c/src/ov_core.cpp

```diff
@@ -132,6 +132,18 @@ ov_status_e ov_core_read_model_from_memory_buffer(const ov_core_t* core,
     return ov_status_e::OK;
 }
 
+ov_status_e ov_core_add_extension(const ov_core_t* core, const char* extension_path) {
+    if (!core || !extension_path) {
+        return ov_status_e::INVALID_C_PARAM;
+    }
+
+    try {
+        core->object->add_extension(extension_path);
+    }
+    CATCH_OV_EXCEPTIONS
+    return ov_status_e::OK;
```

Comment:
The macro `CATCH_OV_EXCEPTIONS` do function return and it looks like this line is dead code.

Reply:
`CATCH_OV_EXCEPTIONS` will return an error status from a caught exception, if no exception occurs then `return ov_status_e::OK;` will be executed, similar to other c-apis in same file.